pkg/mcs/resourcemanager/server/manager.go

```diff
@@ -437,21 +447,94 @@ func (m *Manager) backgroundMetricsFlush(ctx context.Context) {
 					requestCount.DeleteLabelValues(r.name, r.name, writeTypeLabel)
 					availableRUCounter.DeleteLabelValues(r.name, r.name, r.ruType)
 					delete(m.consumptionRecord, r)
+					delete(maxPerSecTrackers, r.name)
+					delete(rruSum, r.name)
+					delete(wruSum, r.name)
+					readRequestUnitMaxPerSecCost.DeleteLabelValues(r.name)
+					writeRequestUnitMaxPerSecCost.DeleteLabelValues(r.name)
 				}
 			}
 		case <-availableRUTicker.C:
 			m.RLock()
+			groups := make([]*ResourceGroup, 0, len(m.groups))
 			for name, group := range m.groups {
 				if name == reservedDefaultGroupName {
 					continue
 				}
+				groups = append(groups, group)
+			}
+			m.RUnlock()
+			// prevent many groups and hold the lock long time.
+			for _, group := range groups {
 				ru := group.getRUToken()
 				if ru < 0 {
 					ru = 0
 				}
-				availableRUCounter.WithLabelValues(name, name).Set(ru)
+				availableRUCounter.WithLabelValues(group.Name, group.Name).Set(ru)
+			}
+
+		case <-recordMaxTicker.C:
+			// Record the sum of RRU and WRU every second.
+			m.RLock()
+			names := make([]string, 0, len(m.groups))
+			for name := range m.groups {
+				names = append(names, name)
 			}
 			m.RUnlock()
+			for _, name := range names {
+				if maxPerSecTrackers[name] == nil {
+					maxPerSecTrackers[name] = newMaxPerSecCostTracker(name, defaultCollectIntervalSec)
+				}
+				maxPerSecTrackers[name].Observe(rruSum[name], wruSum[name])
```

Comment:
```suggestion
				if maxPerSecTrackers[name] == nil && (rruSum[name] + wruSum[name] > 0.0) {
					maxPerSecTrackers[name] = newMaxPerSecCostTracker(name, defaultCollectIntervalSec)
				}
				if maxPerSecTrackers[name] != nil {
					maxPerSecTrackers[name].Observe(rruSum[name], wruSum[name])
				}
```
Maybe skip inactive groups here is better

Reply:
It may be inactive for a period of time, and active for a period of time. To reflect this trend of change, the cost is acceptable